### PR TITLE
Function To scale pods using Kube_Job, DC and attach multiple PVCs

### PR DESCRIPTION
**Pull request by ramkiperiy**
Added Scale lib functions to create pods using kube_job
Added functionality to add mutliple PVCs to single POD
Added functionality to create DC pod
Added functionality to create pod with liveness probe and to start fio IO

Signed-off-by: Ramakrishnan <rperiyas@redhat.com>

### CODE REVIEW

**Comment by ramkiperiy** [comment]
Test execution output run based on non DC pod
`------------------------------------------------------------------------------- live log call -------------------------------------------------------------------------------
19:06:11 - MainThread - test_experian_pvc_creation_deletion - INFO - Start creating 40 PVC of 2 types
19:06:11 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc create on rbd_pvc_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:06:11 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'create', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.rbd_pvc_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:06:15 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc create on cephfs_pvc_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:06:15 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'create', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.cephfs_pvc_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:06:19 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on rbd_pvc_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:06:19 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.rbd_pvc_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:06:24 - MainThread - root - INFO - pvc pvc-test-97eae9e8427f4d1e908d2c902c27ce6 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-c50ee19b878541ce8243232eb217fc4 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-60695fe164a24d0bab556e7a306fe9e status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-bacf3ee4a3604e52ac73f957fb7df5b status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-0a2582fa18fa4f02bec838c9934275d status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-9d50d8cb52f546d68cb768eced45389 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-9b3ca2cac92f439abcd40ecaa5c465d status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-04988bd1611a47f9b50d47f0ee4cdce status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-d412c5b0af1f400ea7469fee43a729b status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-5c3e0b3958f04cedb30e144fa9a50d3 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-7a4f196a84464411a95734d0c40abb6 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-060c2313edf9432cbab019dfa085aef status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-201bc737287e4f8eb2284a612987324 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-e1cb79c2173c4c24b5d5084bbffb42c status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-4b429a673c1643b799932e42a59db5b status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-48302b80005d4884abbed4550a75c62 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-e05c7f510e104c9caadcfe736c48476 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-2603858bce53495f9f65fe7450a28d1 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-fe1398a410f74961928484f71778ae2 status Bound
19:06:24 - MainThread - root - INFO - pvc pvc-test-8758918f656b43f5bf4a5778d7997be status Bound
19:06:24 - MainThread - root - INFO - All PVCs in Bound state
19:06:24 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on cephfs_pvc_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:06:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.cephfs_pvc_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:06:28 - MainThread - root - INFO - pvc pvc-test-67944ef3c2db4575a8e18f5682b34a6 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-213dbb4d4ddc4aeab9eb21e6cb8ee33 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-da751281f8a34bd09278884cbc08482 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-9f5bd32218be40a0a10115593828d12 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-9d575d558a674258b88ce9c3a49b679 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-a8ca6b6d9ade40128b03523d6a50b25 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-227d30845a0742bd943ab347110099e status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-052c345fb05b4a50b6401a320a70944 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-ffaa7fd5135246d7995ed5b4ab501cb status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-f7b7f5eb0baa49c98fcc243658a5f4d status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-2eaaadce2d6d4b70abe9b7ee03c74bc status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-24b4479aafa941acbf77e49e0928f73 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-b5e1d90285ad406984e37ede9d332b7 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-f527e81c483942cda5bf0d249f9c01d status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-989aea032a72476f84183a181cd9173 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-1a61d4a6fb1b4ffab3e02d7706b3ddb status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-1aab98b5e0264a34848afe7d867fb7e status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-9c984a15b3aa49b789e63df63efd4c3 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-70d579e3682c4176aa28bc8a2aa1fa8 status Bound
19:06:28 - MainThread - root - INFO - pvc pvc-test-a0514a42856c4a149b16e366255d95a status Bound
19:06:28 - MainThread - root - INFO - All PVCs in Bound state
19:06:28 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc create on pod_kube_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:06:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'create', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:06:29 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on pod_kube_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:06:29 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:06:30 - MainThread - root - INFO - POD pod-scale-5e800ccdf7554e9996e52b2ca72d19 status Pending
19:06:30 - MainThread - root - INFO - POD pod-scale-8cd395f4f6974d38b3b5fddcd688f6 status Pending
19:08:00 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on pod_kube_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:08:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:08:02 - MainThread - root - INFO - POD pod-scale-5e800ccdf7554e9996e52b2ca72d19 status Running
19:08:02 - MainThread - root - INFO - POD pod-scale-8cd395f4f6974d38b3b5fddcd688f6 status Running
19:08:02 - MainThread - root - INFO - All PODs are in Running state

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/tests/e2e/scale/test_experian_pvc_creation_deletion.py(152)test_2_type_pvc_creation_deletion_scale()
-> job_get_output = job_pod.get(namespace=self.namespace)
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
19:08:38 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on pod_kube_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:08:38 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
19:08:42 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on pod_kube_job object config yaml file in namespace namespace-test-2a4f227ad56046cb803cafb34
19:08:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-21/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-2a4f227ad56046cb803cafb34', '-o', 'yaml']
PASSED      `

Test execution with DC pod
`------------------------------------------------------------------------------- live log call -------------------------------------------------------------------------------
18:58:13 - MainThread - test_experian_pvc_creation_deletion - INFO - Start creating 40 PVC of 2 types
18:58:13 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc create on rbd_pvc_job object config yaml file in namespace namespace-test-ded8124baa9f4ba298ef98bda
18:58:13 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'create', '-f', '/tmp/pytest-of-rperiyas/pytest-20/test_2_type_pvc_creation_delet0/objectconfig.rbd_pvc_job.yaml', '-n', 'namespace-test-ded8124baa9f4ba298ef98bda', '-o', 'yaml']
18:58:16 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc create on cephfs_pvc_job object config yaml file in namespace namespace-test-ded8124baa9f4ba298ef98bda
18:58:16 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'create', '-f', '/tmp/pytest-of-rperiyas/pytest-20/test_2_type_pvc_creation_delet0/objectconfig.cephfs_pvc_job.yaml', '-n', 'namespace-test-ded8124baa9f4ba298ef98bda', '-o', 'yaml']
18:58:21 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on rbd_pvc_job object config yaml file in namespace namespace-test-ded8124baa9f4ba298ef98bda
18:58:21 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-20/test_2_type_pvc_creation_delet0/objectconfig.rbd_pvc_job.yaml', '-n', 'namespace-test-ded8124baa9f4ba298ef98bda', '-o', 'yaml']
18:58:24 - MainThread - root - INFO - pvc pvc-test-85bb9938f7574b9598aa2ca282f1c32 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-607d05cd8ec44f6189f29b73925f1f5 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-d6dd90a2abf14248836db0b3b0aa9a3 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-eca6e6eee2a746d08703c2ca64c1907 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-f297ee10bc6a40f58e44a0ec9f0aed2 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-0a526a47badc4a83a15635282b2cf51 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-85310d6379ad4ba4b948045e0e7e8c4 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-2084d8cc164749c8ae6994310e42b1a status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-1e80b00d9f924bfb9a3248b67c437b4 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-3f13a9f276944daebc2d65f534fe34e status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-0e72afb51a0a49afb76afa927db7500 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-92cc8c8832c040e9b9327feabcf177d status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-f504859493084c799d3e514d6b461ba status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-d5bb225ead364f8ab02ced99887673f status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-6a11a1bf5bd54b4fb2b31977f526521 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-623617e6821d42bb8cfb93a1837e1e0 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-a57afb41f6734d72bd6c8298bd5ff85 status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-0c582bdef32c4b68b4fed64774e865a status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-f7b7822d24f142019255cbfcbc6279d status Bound
18:58:24 - MainThread - root - INFO - pvc pvc-test-d5c5484f768e4c35a6838d8c98b5a62 status Bound
18:58:24 - MainThread - root - INFO - All PVCs in Bound state
18:58:24 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on cephfs_pvc_job object config yaml file in namespace namespace-test-ded8124baa9f4ba298ef98bda
18:58:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-20/test_2_type_pvc_creation_delet0/objectconfig.cephfs_pvc_job.yaml', '-n', 'namespace-test-ded8124baa9f4ba298ef98bda', '-o', 'yaml']
18:58:28 - MainThread - root - INFO - pvc pvc-test-a4b6368763414318883e8608d705955 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-45b7030c90794a68b020791cd4ce852 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-c1923bb4cfea4f2d9a8b1377d3cc513 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-7dec57570f8b4a319bd8d08bdfab3b1 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-f7928edf25f14e8ca051f9477c08409 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-1056ffd29eb84baea85a5582922e938 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-9b67c39234d041c992b937262e6e65f status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-354e3fcfa2094058bf2996c610dbc8a status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-d257304e4fac4598849160620485e3f status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-54b6704616d747bf8b3a70c71e81323 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-c71870cd2013407c9827911bb362545 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-46c9c41a8bb0439989ee86c1bc7c044 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-b20c8f1af79646cab9ce9a673ffb4c3 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-03268a7aabc149e2ab56c3be3d0276b status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-88ecbcd6cc0a435cad4df9be62ee7f0 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-803a0a17f8df4e6cb7ff05cfa6dfb56 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-e9027f0b58f04962a342c52a85b88be status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-d5ed242989834f01acec78b85603294 status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-ccfd334de6d749738675c638b8afe1c status Bound
18:58:28 - MainThread - root - INFO - pvc pvc-test-87d38f9bb7e14adf889505fd424ce77 status Bound
18:58:28 - MainThread - root - INFO - All PVCs in Bound state
18:58:28 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc create on pod_kube_job object config yaml file in namespace namespace-test-ded8124baa9f4ba298ef98bda
18:58:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'create', '-f', '/tmp/pytest-of-rperiyas/pytest-20/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-ded8124baa9f4ba298ef98bda', '-o', 'yaml']
18:58:30 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on pod_kube_job object config yaml file in namespace namespace-test-ded8124baa9f4ba298ef98bda
18:58:30 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-20/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-ded8124baa9f4ba298ef98bda', '-o', 'yaml']

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/scale_lib.py(1033)check_all_pod_reached_running_state_in_kube_job()
-> for i in range(no_of_pod):
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
18:59:34 - MainThread - root - INFO - DC Config pod-scale-7cad96f2963240b7ace14cc928ec84 available running pods 0
18:59:34 - MainThread - root - INFO - DC Config pod-scale-0dccb422d3ad42e1a84b151ed08696 available running pods 0
19:01:04 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on pod_kube_job object config yaml file in namespace namespace-test-ded8124baa9f4ba298ef98bda
19:01:04 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-20/test_2_type_pvc_creation_delet0/objectconfig.pod_kube_job.yaml', '-n', 'namespace-test-ded8124baa9f4ba298ef98bda', '-o', 'yaml']

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/ocs_ci/ocs/scale_lib.py(1032)check_all_pod_reached_running_state_in_kube_job()
-> if job_get_output["items"][0]["kind"] == constants.POD:
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
19:01:11 - MainThread - root - INFO - DC Config pod-scale-7cad96f2963240b7ace14cc928ec84 available running pods 1
19:01:11 - MainThread - root - INFO - DC Config pod-scale-0dccb422d3ad42e1a84b151ed08696 available running pods 1
19:01:11 - MainThread - root - INFO - All PODs are in Running state

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/experien/ocs-ci/tests/e2e/scale/test_experian_pvc_creation_deletion.py(152)test_2_type_pvc_creation_deletion_scale()
-> job_get_output = job_pod.get(namespace=self.namespace)
`

**Comment by ramkiperiy** [comment]
Fix for https://github.com/red-hat-storage/ocs-ci/issues/2744

File: ocs_ci/ocs/scale_lib.py [hunk 1]
```diff
@@ -998,3 +1000,229 @@ def get_max_pvc_count():
...
+                    )
+
+        # Check the length of pod_not_running_list to decide either all PODs reached
+        # Running state, If not then wait for 30secs and re-iterate while loop
+        if len(pod_not_running_list):
+            time.sleep(timeout)
+            while_iteration_count += 1
+            # Breaking while loop after 10 Iteration i.e. after 30*10 secs of wait_time
+            # And if PODs are still not in bound state then there will be assert.
+            if while_iteration_count >= 10:
```
Comment (Avilir) [inline ocs_ci/ocs/scale_lib.py:1066]: i think this should depend on the number of pods, for instance,  1500 pods will take more time then 120 pods, and less then 10K pods

Comment (ramkiperiy) [reply ocs_ci/ocs/scale_lib.py:1066]: Pods will be created using kube_job so in TC if we are creating 3000 PVC's and if we attach 30 PVCs to single pod then we will end up creating just 100 pods.
Will test one more time and update here to with results of 3000, 6000 PVCs creation.

File: ocs_ci/ocs/scale_lib.py [hunk 2]
```diff
@@ -998,3 +1000,251 @@ def get_max_pvc_count():
...
+        elif not len(pod_not_running_list):
+            for i in range(no_of_pod):
+                pod_running_list.append(job_get_output["items"][i]["metadata"]["name"])
+            logging.info("All PODs are in Running state")
+            break
+
+    return pod_running_list
+
+
+def attach_multiple_pvc_to_pod_dict(
```
Comment (ramkiperiy) [inline ocs_ci/ocs/scale_lib.py:1083]: Sample yaml file generated by this function can be found in http://pastebin.test.redhat.com/954350

File: ocs_ci/ocs/scale_lib.py [hunk 3]
```diff
@@ -998,3 +1000,251 @@ def get_max_pvc_count():
...
+                    template_list = {
+                        "template": {"metadata": {"labels": {"name": pod_name}}}
+                    }
+                    pod_data["spec"] = template_list
+                    pod_data["spec"]["template"]["spec"] = spec_containers
+                    pod_data["spec"]["template"]["spec"]["restartPolicy"] = "Always"
+                    pod_data["spec"]["template"]["spec"]["containers"][0]["args"] = [
+                        "/bin/sh",
+                        "-c",
+                        f"fio --name=fio-rand-readwrite --filename={liveness_check_path}/abc "
```
Comment (keesturam) [inline ocs_ci/ocs/scale_lib.py:1168]: This is a good start. 
maybe in a subsequent PR, We should try to have a variety of IOs.

File: ocs_ci/ocs/scale_lib.py [hunk 4]
```diff
@@ -998,3 +1000,251 @@ def get_max_pvc_count():
...
+        temp_list.append(pvc_name)
+        if len(temp_list) == pvcs_per_pod:
+            pod_dict = constants.PERF_POD_YAML
+            pod_data = templating.load_yaml(pod_dict)
+            pod_name = helpers.create_unique_resource_name("scale", "pod")
+
+            # Update pod yaml with required params
+            pod_data["metadata"]["name"] = pod_name
+            pod_data["metadata"]["namespace"] = namespace
+            volume_list = pod_data.get("spec").get("volumes")
```
[the remark this answers is not in the document]
Comment (ramkiperiy) [reply ocs_ci/ocs/scale_lib.py:1120]: But the first value is not required, also it comes from default pod yaml file.

**Comment by ramkiperiy** [comment]
Complete TC execution logs can be found in https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/1866/console, this is from Jenkins execution of PR #4111  which uses the functions.

File: ocs_ci/ocs/scale_lib.py [hunk 5]
```diff
@@ -998,3 +1000,251 @@ def get_max_pvc_count():
...
+
+    """
+    job_get_output = kube_job_obj.get(namespace=namespace)
+    pod_dict = dict()
+    for i in range(no_of_pod):
+        started_at_str = job_get_output["items"][i]["status"]["containerStatuses"][0][
+            "state"
+        ]["running"]["startedAt"]
+        start_time_str = job_get_output["items"][i]["status"]["startTime"]
+        started_at = re.search(r"(\d\d):(\d\d):(\d\d)", started_at_str)
```
Comment (vasukulkarni) [inline ocs_ci/ocs/scale_lib.py:1239]: I think this could be moved out to utils since we have few places where we use this logic, can be done in next PR.

**Review by vasukulkarni — APPROVED** [review]

**Review by tiffanyn108 — APPROVED** [review]
LGTM.

File: ocs_ci/ocs/scale_lib.py (hunk 5 above)
Comment (ramkiperiy) [reply ocs_ci/ocs/scale_lib.py:1239]: Sure @vasukulkarni we can enhance it

**Pull request merged**